src/main/java/org/manifold/compiler/front/TupleValueVertex.java

```diff
@@ -61,9 +60,11 @@ public String toString() {
   @Override
   public void elaborate() throws Exception {
     log.debug("elaborating tuple value");
-    Map<String, Value> values = new HashMap<>();
-    Map<String, TypeValue> types = new HashMap<>();
-    for (Map.Entry<String, ExpressionEdge> entry : valueEdges.entrySet()) {
+
+    MappedArray<String, Value> values = new MappedArray<>();
+    // TODO(nikklassen) this should also be a MappedArray
+    Map<String, TypeValue> types = new LinkedHashMap<>();
```

Comment:
It shouldn't take too long, but I'm not sure I'll get time to look at it in the next two weeks.